docs/source/en/task_summary.mdx

```diff
@@ -10,1125 +10,222 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
+There are two types of language modeling:
 
-The following examples demonstrate how to use a [`pipeline`] and a model and tokenizer for image classification inference:
+* causal: the model's objective is to predict the next token in a sequence, and future tokens are masked
+* masked: the model's objective is to predict a masked token in a sequence with full access to the tokens in the sequence
 
 ```py
 >>> from transformers import pipeline
+# causal language modeling
 
->>> vision_classifier = pipeline(task="image-classification")
->>> result = vision_classifier(
-...     images="https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
-... )
->>> print("\n".join([f"Class {d['label']} with score {round(d['score'], 4)}" for d in result]))
-Class lynx, catamount with score 0.4335
-Class cougar, puma, catamount, mountain lion, painter, panther, Felis concolor with score 0.0348
-Class snow leopard, ounce, Panthera uncia with score 0.0324
-Class Egyptian cat with score 0.0239
-Class tiger cat with score 0.0229
-```
-
-The general process for using a model and image processor for image classification is:
-
-1. Instantiate an image processor and a model from the checkpoint name.
-2. Process the image to be classified with an image processor.
-3. Pass the input through the model and take the `argmax` to retrieve the predicted class.
-4. Convert the class id to a class name with `id2label` to return an interpretable result.
-
-<frameworkcontent>
-<pt>
-```py
->>> from transformers import AutoImageProcessor, AutoModelForImageClassification
->>> import torch
->>> from datasets import load_dataset
-
->>> dataset = load_dataset("huggingface/cats-image")
->>> image = dataset["test"]["image"][0]
-
->>> feature_extractor = AutoImageProcessor.from_pretrained("google/vit-base-patch16-224")
->>> model = AutoModelForImageClassification.from_pretrained("google/vit-base-patch16-224")
+>>> prompt = "Hugging Face is a"
+>>> text_generator = pipeline(task="text-generation")
+>>> text_generator(prompt)
+# masked language modeling
 
->>> inputs = feature_extractor(image, return_tensors="pt")
-
->>> with torch.no_grad():
-...     logits = model(**inputs).logits
-
->>> predicted_label = logits.argmax(-1).item()
->>> print(model.config.id2label[predicted_label])
-Egyptian cat
+>>> text = "Hugging Face is a <mask> company based in New York City."
+>>> fill_mask = pipeline(task="fill-mask")
+>>> fill_mask(text, top_k=3)
 ```
-</pt>
-</frameworkcontent>
+
+Hopefully, this has given you some more background information about all these tasks and the practical importance of each one. The Transformer really changed the paradigm for how we approach all these different tasks and it's exciting to see how Transformer models are being used to solve problems new and old. In the next section, you'll learn **how** 🤗 Transformers work to solve these tasks.
```

Comment:
Will the "next section" be part of this document or do you plan to have a separate doc for the "how"?

Reply:
It'll be a separate [doc](https://github.com/huggingface/transformers/pull/21067) for the "how".